src/xmpp/dns.go

```diff
@@ -27,7 +27,18 @@ func HomeServerAddrs(jid JID) (addr []string, err error) {
 
 	// Build list of "host:port" strings.
 	for _, a := range addrs {
-		addr = append(addr, fmt.Sprintf("%s:%d", a.Target, a.Port))
+		target := parseTargetDomainName(a.Target)
+		addr = append(addr, fmt.Sprintf("%s:%d", target, a.Port))
+	}
+	return
+}
+
+// Remove the last dot in the domain name if exist
+func parseTargetDomainName(domainName string) (ret string) {
+	if domainName[len(domainName)-1] == '.' {
```

Comment:
If I remember well, it cause me some troubles, but can't reproduce. I didn't know about `strings.TrimRight()` which is better than what I've done. Will change that.